src/drive/web/modules/navigation/RealtimeFiles.spec.jsx

```diff
@@ -31,86 +30,58 @@ describe('RealTimeFiles', () => {
-    expect(realtime.subscribe).toHaveBeenCalledWith(
+    const files = [
       {
-        token: '1',
-        url: 'http://mycozy.cloud'
+        id: '2',
+        _id: '2'
       },
-      'io.cozy.files'
-    )
-  })
-
-  it('should instanciate Realtime and update when context change', () => {
-    const context = {
-      client: {
-        stackClient: {
-          token: {
-            token: '1'
-          },
-          uri: 'http://mycozy.cloud'
-        }
-      }
-    }
-    const component = mount(<RealtimeFiles />, { context })
-    // jest.spyOn(realtime, 'subscribe')
-    expect(realtime.subscribe).toHaveBeenCalledWith(
       {
-        token: '1',
-        url: 'http://mycozy.cloud'
-      },
-      'io.cozy.files'
-    )
-
-    component.setContext({
-      client: {
-        stackClient: {
-          token: {
-            token: '2'
-          },
-          uri: 'http://mycozy.cloud'
-        }
+        id: '3',
+        _id: '3'
       }
-    })
-    expect(realtime.subscribe).toHaveBeenCalledWith(
-      {
-        token: '2',
-        url: 'http://mycozy.cloud'
-      },
-      'io.cozy.files'
+    ]
+    const updateFile = jest.fn()
+    const updateOfflineFileCopyIfNecessary = jest.fn()
+    const addFile = jest.fn()
+    const deleteFile = jest.fn()
+    const component = mount(
+      <RealtimeFiles
+        client={client}
+        files={files}
+        updateFile={updateFile}
+        addFile={addFile}
+        deleteFile={deleteFile}
+        // location={{ pathname: 'recent' }}
+        updateOfflineFileCopyIfNecessary={updateOfflineFileCopyIfNecessary}
+      />
     )
-  })
+    //Doc was present and in the current view => updateFile
+    component.instance().isInCurrentView = jest.fn().mockReturnValue(true)
 
-  it('should instanciate Realtime even with accessToken ', () => {
-    const context = {
-      client: {
-        stackClient: {
-          token: {
-            accessToken: '1'
-          },
-          uri: 'http://mycozy.cloud'
-        }
-      }
-    }
-    mount(<RealtimeFiles />, { context })
-    // jest.spyOn(realtime, 'subscribe')
-    expect(realtime.subscribe).toHaveBeenCalledWith(
-      {
-        token: '1',
-        url: 'http://mycozy.cloud'
-      },
-      'io.cozy.files'
-    )
+    component.instance().onDocumentChange({ _id: '2' })
+    expect(updateFile).toBeCalled()
+    expect(updateOfflineFileCopyIfNecessary).toBeCalled()
+
+    //Doc was present, but not the current view => delete
+    component.instance().isInCurrentView = jest
+      .fn()
+      .mockReturnValueOnce(false)
+      .mockReturnValue(true)
+    component.instance().onDocumentChange({ _id: '2' })
+    expect(deleteFile).toBeCalled()
+
+    //Doc was not present, but the current view => add
+    component.instance().isInCurrentView = jest.fn().mockReturnValue(true)
+    component.instance().onDocumentChange({ _id: '4' })
+    expect(addFile).toBeCalled()
```

Comment:
There's a potential issue that these test don't cover: if the update test also triggers a call to `deleteFile`, it won't be detected. I think the safest way to test these would be to reset the mocks after each test, and to also test stuff like `expect(deleteFile).not.toBeCalled()`.